src/main/scala/mesosphere/marathon/core/launcher/TaskOp.scala

```diff
@@ -37,13 +40,65 @@ object TaskOp {
 
   case class ReserveAndCreateVolumes(
       newTask: Task,
-      resources: Iterable[Mesos.Resource],
+      resources: Iterable[MesosProtos.Resource],
       localVolumes: Iterable[LocalVolume],
       oldTask: Option[Task] = None,
-      offerOperations: Iterable[Mesos.Offer.Operation]) extends TaskOp {
+      offerOperations: Iterable[MesosProtos.Offer.Operation]) extends TaskOp {
+
+    override def taskId: Task.Id = newTask.taskId
+    override def maybeNewTask: Option[Task] = Some(newTask)
 
-    override def applyToOffer(offer: Mesos.Offer): Mesos.Offer =
+    override def applyToOffer(offer: MesosProtos.Offer): MesosProtos.Offer =
       ResourceUtil.consumeResourcesFromOffer(offer, resources)
   }
 
+  case class UnreserveAndDestroyVolumes(
+      taskId: Task.Id,
+      maybeNewTask: Option[Task] = None,
+      resources: Iterable[MesosProtos.Resource],
+      oldTask: Option[Task] = None) extends TaskOp {
+
+    override lazy val offerOperations: Iterable[MesosProtos.Offer.Operation] = {
+      val (withDisk, withoutDisk) = resources.partition(_.hasDisk)
```

Comment:
Do we need to filter resources that belong to us, or is this list already filtered?